wire-ios-sync-engine/Tests/Source/SessionManager/SessionManagerTests.swift

```diff
@@ -400,6 +400,31 @@ final class SessionManagerTests: IntegrationTest {
         }
     }
 
+    func tmpDirectoryPath() -> URL {
+        return URL(fileURLWithPath: NSTemporaryDirectory())
+    }
+
+    func testThatItDestroyedTmpDirectoryAfterLoggedOut() {
+
+        // GIVEN
+        XCTAssertTrue(login())
+        let observer = SessionManagerObserverMock()
+        let token = sessionManager?.addSessionManagerDestroyedSessionObserver(observer)
+
+        var tempUrl = self.tmpDirectoryPath().appendingPathComponent("testFile.txt")
+        let testData = "Test Message"
+        try? testData.write(to: tempUrl, atomically: true, encoding: .utf8)
```

Comment:
As suggested, asserted temp directory when it Is not empty.

Reply:
I don't see the assert here. I see the assert at the end that the temp directory is empty. But I was suggesting to assert the temp director is not empty:

```swift
// Given
let tempDirectory = ...
let file = ...
tempDirectory.addFile(file)
XCTAssertFalse(tempDirectory.isEmpty)

// When
...

// Then
...

```

The reason is that if we failed to set up the test and not add any file to the temp directory, then it wouldn't matter what happens in the WHEN part because the assertion that the temp directory is empty would always be true. By adding this assertion in GIVEN we can see if we failed to set up the test.

Reply:
As suggested, assert is added

Reply:
`XCTAssertFalse(tempUrl.path.isEmpty)` looks like it's asserting that the path (a string maybe?) is not empty. Instead we should assert like you do below:

```
XCTAssertFalse(FileManager.default.fileExists(atPath: tempUrl.path))
```

Reply:
It is added already, if you notice the entire method, this assert already exists.

Reply:
But not on line 417. We want to assert that there is some data in the temp url, not asser that the path to the temp url is non empty.